### PR TITLE
Provision ES storage implementation based on configured ES version.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This change introduces a quick and dirty mechanism to configure the ES storage implementation used. The user is now able to define a `elasticsearch_version` configuration directive where just the major version (for now) is specified. Every storage implementation can register its adapters to a version. At runtime, the configured version is used to resolve interface bindings to their corresponding implementations.

This implementation is a bit ugly at the moment. It requires quite some boiler plate, but works for now, especially considering that the number of interfaces is not too big, reducing the boilerplate.

One way to improve this in the future is to load storage driver modules differently instead of reusing our generic plugin mechanism. This would allow us to define a different interface for storage driver modules, exposing the version they support and the module they need for binding explicitly. With this information, we could bind only the driver module which we want (and only once), instead of resolving at runtime.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java [hunk 1]
```diff
@@ -97,8 +97,20 @@ protected void configure() {
         registerPivotBucketHandler(Time.NAME, ESTimeHandler.class);
         registerPivotBucketHandler(DateRangeBucket.NAME, ESDateRangeHandler.class);
 
-        bind(ExportBackend.class).to(ElasticsearchExportBackend.class);
-        bind(RequestStrategy.class).to(org.graylog.storage.elasticsearch6.views.export.Scroll.class);
+        bindExportBackend().to(ElasticsearchExportBackend.class);
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java:100]: Sure, I did that before, I just consider this to be clearer regarding its intention.

[the remark this answers is not in the document]
Comment (alex-konn) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java:100]: Yeah, I understand. Keeping the `configure` method tidy and easy to read is important.

I guess the only thing that really confused me was that that one of these two methods was in the base class `ViewsModule`. I would personally get rid of at least that one and move the call to `bindForVersion` into `ViewsESBackendModule`, but this is nothing that would keep me from merging this PR. :D

[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java:100]: `registerQueryBackend(Version, String, ...)` is supposed to be reused in future storage modules, that's why it is in `ViewsModule`.

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/Elasticsearch6Plugin.java [hunk 2]
```diff
@@ -4,10 +4,13 @@
 import org.graylog2.plugin.Plugin;
 import org.graylog2.plugin.PluginMetaData;
 import org.graylog2.plugin.PluginModule;
+import org.graylog2.plugin.Version;
 
 import java.util.Collection;
 
 public class Elasticsearch6Plugin implements Plugin {
+    public static final Version SUPPORTED_VERSION = Version.from(6, 0, 0);
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/Elasticsearch6Plugin.java:12]: :heavy_check_mark:

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex-konn) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ViewsESBackendModule.java:100]: Alright, then let's leave it as it is.

**Pull request merged**